### PR TITLE
k8s yaml changes + addition of config file for instrumentation

### PR DESCRIPTION
**Pull request by LoekL**
This PR contains the changes to k8s YAML files to enable the analytics instrumentation, and adds an analytics configmap.

### CODE REVIEW

**Comment by LoekL** [comment]
@JoelOtter - FYI I upgraded all ESP container versions to 1.38.0 & verified everything is still working.

Means after we merge this in we can close-out: https://improbableio.atlassian.net/browse/OS-575

File: services/k8s/analytics-config.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "match":
+      "party_matching":
+        disabled: true
+      "party_requeued":
+        disabled: true
+      "player_matching":
+        disabled: true
+      "player_requeued":
+        disabled: true
+kind: ConfigMap
```
Comment (JoelOtter) [inline services/k8s/analytics-config.yaml]: Can you put the `kind` and `metadata` fields above `data`, please?

File: services/k8s/analytics-config.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+apiVersion: v1
+data:
+  analytics-config.yaml: |-
```
Comment (JoelOtter) [inline services/k8s/analytics-config.yaml]: what does the `|-` syntax do here?

File: services/k8s/config.yaml [hunk 3]
```diff
@@ -5,5 +5,5 @@ metadata:
   selfLink: /api/v1/namespaces/default/configmaps/online-services-config
 data:
   redis.connection-string: "[your Redis host]:6379"
-  spatialos.project: "[your spatialOS project]"
+  spatialos.project: "[your SpatialOS project id]"
```
Comment (JoelOtter) [inline services/k8s/config.yaml]: We should probably make these match the others, like `{{squiggly_and_lowercase}}`. Have a look at the other files in master. If they're not in `service-analytics` you need to rebase it, I'm afraid.

**Comment by JoelOtter** [comment]
@LoekL you need to run `dotnet format`, looks like

File: services/k8s/analytics-config.yaml (hunk 1 above)
Comment (LoekL) [reply services/k8s/analytics-config.yaml]: Done.

File: services/k8s/config.yaml (hunk 3 above)
Comment (LoekL) [reply services/k8s/config.yaml]: Done.

File: services/k8s/analytics-config.yaml (hunk 2 above)
Comment (LoekL) [reply services/k8s/analytics-config.yaml]: See Stripping [here](https://adminswerk.de/multi-line-string-yaml-ansible-I/).

File: services/csharp/Party.Test/KickOutPlayerShould.cs [hunk 4]
```diff
@@ -164,7 +164,7 @@ public void ReturnPermissionDeniedWhenInitiationAndEvictedAreNotMembersOfTheSame
             _mockMemoryStoreClient.Setup(client => client.GetAsync<PartyDataModel>(_testParty.Id))
                 .ReturnsAsync(_testParty);
 
-            // Check that the kick-out request will throw an exception. 
+            // Check that the kick-out request will throw an exception.
             var context = Util.CreateFakeCallContext(TestInitiatorPlayerId, Pit);
             var exception =
                 Assert.ThrowsAsync<RpcException>(() => _partyService.KickOutPlayer(_testKickOutRequest, context));
```
Comment (LoekL) [inline services/csharp/Party.Test/KickOutPlayerShould.cs]: I somehow can't revert the above removal of trailing whitespaces..

File: services/k8s/deployment-pool/deployment.yaml [hunk 5]
```diff
@@ -22,14 +22,24 @@ spec:
...
               name: "launch-config"
+      - name: analytics-api-key
+        secret:
+          secretName: analytics-api-key
+      - name: analytics-config
+        configMap:
+          name: analytics-config
       containers:
       - name: deployment-pool
-        image: gcr.io/{{your_google_project_name}}/deployment-pool
```
Comment (LoekL) [inline services/k8s/deployment-pool/deployment.yaml]: project name is simply wrong, it's project id

File: services/csharp/Party.Test/KickOutPlayerShould.cs (hunk 4 above)
Comment (JoelOtter) [reply services/csharp/Party.Test/KickOutPlayerShould.cs]: Just run `dotnet format` over the whole thing, it'll fix anything that needs fixing

**Pull request merged**